File: app/src/main/java/com/hijamoya/example/dagger/injection/ActivityBindingModule.java
package com.hijamoya.example.dagger.injection;

import android.app.Activity;

import com.hijamoya.example.ui.MainActivity;
import com.hijamoya.example.ui.MainActivityComponent;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

@Module(subcomponents = {MainActivityComponent.class})
public abstract class ActivityBindingModule {

    @Binds
    @IntoMap
    @ActivityKey(MainActivity.class)
    abstract AndroidInjector.Factory<? extends Activity>
    bindYourActivityInjectorFactory(MainActivityComponent.Builder builder);

}

File: app/src/main/kotlin/com/hijamoya/example/dagger/injection/ActivityBindingModule.kt
package com.hijamoya.example.dagger.injection

import com.hijamoya.example.ui.MainActivity
import com.hijamoya.example.ui.MainActivityComponent
import dagger.Binds
import dagger.Module
import dagger.android.AndroidInjector
import dagger.multibindings.ClassKey
import dagger.multibindings.IntoMap

@Module(subcomponents = [MainActivityComponent::class])
abstract class ActivityBindingModule {

    @Binds
    @IntoMap
    @ClassKey(MainActivity::class)
    internal abstract fun bindYourAndroidInjectorFactory(factory: MainActivityComponent.Factory):
            AndroidInjector.Factory<*>
}
